src/parser/demonhunter/havoc/modules/spells/azeritetraits/ThirstingBlades.js

```diff
@@ -0,0 +1,105 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { calculateAzeriteEffects } from 'common/stats';
+import SPELLS from 'common/SPELLS';
+import Events from 'parser/core/Events';
+import AzeritePowerStatistic from 'interface/statistics/AzeritePowerStatistic';
+import SpellLink from 'common/SpellLink';
+import { formatNumber, formatPercentage } from 'common/format';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import React from 'react';
+
+// grabbed from RevolvingBlades.js
+const azeriteTraitStats = traits => Object.values(traits).reduce((obj, rank) => {
+  const [damage] = calculateAzeriteEffects(SPELLS.THIRSTING_BLADES.id, rank);
+  obj.damage += damage;
+  return obj;
+}, {
+  damage: 0,
+});
+
+/**
+ * Thirsting Blades Azerite Power
+ * Requires Demon Hunter (Havoc)
+ * 
+ * Every 1.5 sec, increase the damage of Chaos Strike by 10 and reduce its cost by 1 Fury. This effect stacks
+ * 
+ * Heavily borrowed from RevolvingBlades.js to make this module
+ */
+class ThirstingBlades extends Analyzer {
+
+  damagePerThirstingBlades = 0; // calculate damage per stack for azerite trait
+  stacksGained = 0; // total over entire fight used (wont count stacks that dont get used like at end of fight)
+  castedCount = 0; // how many chaos strikes or annhilation casts (buff is removed/used)
+  currentStacks = 0; // tracks the stack amount everytime the buff is removed/used
+
+  constructor(...args) {
+    super(...args);
+
+    this.addEventListener(Events.applybuffstack.by(SELECTED_PLAYER), this.onApplyBuffStack);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER), this.onApplyBuff);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER), this.onRemoveThirstingBlades);
+
+    const { damage } = azeriteTraitStats(this.selectedCombatant.traitsBySpellId[SPELLS.THIRSTING_BLADES.id]);
+    this.damagePerThirstingBlades = damage;
+
+  }
+
+  onApplyBuff(event) {
+    if(event.ability.name === "Thirsting Blades") {
```

Comment:
```suggestion
    if(event.ability.guid === SPELLS.THIRSTING_BLADES.id) {
```
We usually check for abilities by using their ID as that won't run into issues like uniqueness. 

You can alternatively attach to the spell you're looking for to the event listener itself 
(like `this.addEventListener(Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.THIRSTING_BLADES), this.onApplyBuffStack);`
instead of
`this.addEventListener(Events.applybuffstack.by(SELECTED_PLAYER), this.onApplyBuffStack);` in row 38)
and then you don't have to check for the correct spell at all.

Reply:
fixed this.